File: src/services/tauri/performance-commands.ts
import { invoke } from "@tauri-apps/api/core";
import { logger } from "@/lib/logger";
import type { CommandResult } from "./core";
import type { PerformanceConfig } from "@/types/settings";

/**
 * Get the current performance configuration
 */
export async function getPerformanceConfig(): Promise<CommandResult<PerformanceConfig>> {
  try {
    const result = await invoke<PerformanceConfig>("get_performance_config");
    logger.info("Performance config retrieved", result);
    return { success: true, data: result };
  } catch (error) {
    logger.error("Failed to get performance config", error);
    return {
      success: false,
      error: error instanceof Error ? error.message : String(error),
    };
  }
}

/**
 * Update performance configuration
 * @param config The new performance configuration
 * @param persist Whether to persist the changes to disk (default: false for session-only)
 */
export async function updatePerformanceConfig(
  config: PerformanceConfig,
  persist: boolean = false
): Promise<CommandResult<PerformanceConfig>> {
  try {
    const result = await invoke<PerformanceConfig>("update_performance_config", {
      config,
      persist,
    });
    logger.info("Performance config updated", { config, persist });
    return { success: true, data: result };
  } catch (error) {
    logger.error("Failed to update performance config", error);
    return {
      success: false,
      error: error instanceof Error ? error.message : String(error),
    };
  }
}

File: src-tauri/src/commands/performance_commands.rs
use tauri::{AppHandle, State};

use crate::{app_state::PerformanceConfigState, PerformanceConfig};

/// Get the current performance configuration
#[tauri::command]
#[allow(dead_code)]
pub(crate) fn get_performance_config(
    perf_config: State<'_, PerformanceConfigState>,
) -> Result<PerformanceConfig, String> {
    crate::application::performance::get_performance_config(&perf_config)
}

/// Update performance configuration
#[tauri::command]
#[allow(dead_code)]
pub(crate) async fn update_performance_config(
    app: AppHandle,
    perf_config: State<'_, PerformanceConfigState>,
    config: PerformanceConfig,
    persist: bool,
) -> Result<PerformanceConfig, String> {
    crate::application::performance::update_performance_config(&app, &perf_config, config, persist)
        .await
}
